rlgear/utils.py

```diff
@@ -262,6 +262,21 @@ def shorten_dfs(_dfs: Sequence[pd.DataFrame], max_step: int = None) -> None:
         _dfs[i] = _df[_df.index <= max_step]  # type: ignore
 
 
+def preprocess_pbt_df(df: pd.DataFrame, x_tag: str) -> pd.DataFrame:
```

Comment:
I had a bit of difficulty following the logic. Can you add a docstring or comments?

Reply:
added a docstring and comments